lib/pure/sugar.nim

```diff
@@ -156,16 +156,45 @@ macro dump*(x: untyped): untyped =
   ## It accepts any expression and prints a textual representation
   ## of the tree representing the expression - as it would appear in
   ## source code - together with the value of the expression.
+  ##
+  ## Deprecated: use `dumpToString` which is more convenient and useful since
```

Comment:
`dump` is not "deprecated" just because dumpToString exists now which does something else. This makes no sense. So why is `dump` not `echo + dumpToString` then?

Reply:
I've rephrased to avoid the word "Deprecated", future work may change implementation of dump to be `echo + dumpToString` or similar, and where it should be marked as deprecated can be discussed there instead of here